pkg/services/internal/whatsonchain/service.go

```diff
@@ -141,3 +135,37 @@ func (woc *WhatsOnChain) UpdateBsvExchangeRate() (defs.BSVExchangeRate, error) {
 
 	return woc.bsvExchangeRate, nil
 }
+
+func (woc *WhatsOnChain) FindChainTipHeader(ctx context.Context) (*wdk.ChainBlockHeader, error) {
+	var results dto.BlockHeaders
+	url := fmt.Sprintf("%s/block/headers?limit=1", woc.url)
+	res, err := woc.
+		httpClient.
+		R().
+		SetContext(ctx).
+		SetResult(&results).
+		AddRetryCondition(func(res *resty.Response, err error) bool {
+			return res.StatusCode() == http.StatusTooManyRequests
+		}).
+		Get(url)
+
+	if err != nil {
+		return nil, fmt.Errorf("error while fetching block headers from WhatsOnChain (URL: %s): %w", url, err)
+	}
+
+	if res.StatusCode() != http.StatusOK {
+		return nil, fmt.Errorf("unexpected response from WhatsOnChain (URL: %s): status code %d", url, res.StatusCode())
+	}
+
+	if results.IsEmpty() {
+		return nil, fmt.Errorf("no block headers returned from WhatsOnChain (URL: %s); at least one expected", url)
+	}
+
+	first := results.First()
```

Comment:
IMHO it's not really needed to have this First method, because you already check if it is empty.

Eventually you could change the method First, to use `optional` package
```go
func (bb BlockHeaders) First() optional.Value[BlockHeader] {
	if bb.IsEmpty() {
		return optional.Empty[BlockHeader]()
	}
	return optional.OfValue(bb[0])
}
```

and then this part could be changed to:
```suggestion
	first, err := results.First().OrError(fmt.Errorf("no block headers returned from WhatsOnChain (URL: %s); at least one expected", url))
	if err != nil {
	  return nil, err
	}
```

Aha, this approach would need a small adjustement of .golangci-lint.yml by adding:
```yaml
linters-settings:
  wrapcheck:
    extra-ignore-sigs:
      - .OrError(err error)
```

Reply:
As discussed during chat, we'll use the straightforward approach (plain Go).